Aleo schnorr

anstylian opened the pull request.

anstylian committed:
schnorr signature

anstylian committed:
clean up, pending: generate aleo key correctly

anstylian committed:
update proto

anstylian committed:
use tofn

tilacog commented on Cargo.toml:
```diff
@@ -46,6 +46,10 @@ zeroize = { version = "1.8", features = ["zeroize_derive"], default-features = f
 # error handling
 thiserror = { version = "1.0", default-features = false }
 anyhow = { version = "1.0", default-features = false }
+snarkvm = { version = "1.1.0" } # TODO: include only the features we need
+# snarkvm = { version = "1.1.0", default-features =  false, features = ["parameters"]}
```
What's the context for this commented out entry?

tilacog replied:
Also, it seems this toml file has sections delimited by comments, and our new entries are right under `# error handling`. Can we follow the original structure and use/create a dedicated section too?

tilacog commented on src/multisig/mod.rs:
```diff
@@ -1,3 +1,4 @@
+// mod aleo_schnorr_signature;
```
Should we remove this comment?

tilacog reviewed (approved):
Left a few comments, but didn't see nothing major, so feel free after discussing/addressing them.

On a side note, I observed this code uses our unmerged `tofn` branch:
https://github.com/axelarnetwork/tofn/compare/main...eigerco:tofn:aleo-schnorr

There's an entire test commented out there. Should we worry about it?

anstylian committed:
review changes

anstylian committed:
review changes

anstylian committed:
update tofn

anstylian committed:
refactor: Aleo schnoor support

anstylian committed:
apply clippy fix

anstylian committed:
remove a by-pass fix

anstylian committed:
refactor: upgrade to snarkvm 4.1.0 with modular dependencies

Replace monolithic snarkvm 3.7.1 with targeted snarkvm 4.1.0 modules
to reduce dependency bloat and improve build times.

- Replace snarkvm with snarkvm-console-network for core functionality
- Add snarkvm-console-account and snarkvm-utilities for test support
- Update network type imports to use modular console paths
- Fix Aleo Schnorr test message to use valid group element
- Remove 180+ unused transitive dependencies from lock file

This maintains full API compatibility while significantly reducing
the dependency footprint from a full SnarkVM installation to only
the required console components.

anstylian committed:
simplify aleo signature for test

anstylian committed:
allow exactly one aleo network to be allowed. Remove hardcoded network for testing

anstylian committed:
Merge pull request #4 from eigerco/refactor-aleo-schnorr

Refactor aleo-schnorr support

anstylian committed:
update snarkvm

anstylian committed:
update snarkvm pointer to snarkvm-with-cosmwasm tag, and update tofn pointer to latest update

anstylian committed:
Merge pull request #5 from eigerco/aleo-schnorr-update

update snarkvm

anstylian committed:
Aleo schnorr update (#6)

* patch snarkvm-algorithms
* patch snarkvm-console-account
* patch snarkvm-utilities
